qiskit/algorithms/gradients/finite_diff_estimator_gradient.py

```diff
@@ -0,0 +1,99 @@
+
+import numpy as np
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.exceptions import QiskitError
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.providers import JobStatus
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .base_estimator_gradient import BaseEstimatorGradient
+from .estimator_gradient_result import EstimatorGradientResult
+
+
+class FiniteDiffEstimatorGradient(BaseEstimatorGradient):
+    """
+    Compute the gradients of the expectation values by finite difference method.
+    """
+
+    def __init__(self, estimator: BaseEstimator, epsilon: float, **run_options):
+        """
+        Args:
+            estimator: The estimator used to compute the gradients.
+            epsilon: The offset size for the finite difference gradients.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in ``run`` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+
+        Raises:
+            ValueError: If ``epsilon`` is not positive.
+        """
+        if epsilon <= 0:
+            raise ValueError(f"epsilon ({epsilon}) should be positive.")
+        self._epsilon = epsilon
+        self._base_parameter_values_dict = {}
+        super().__init__(estimator, **run_options)
+
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None],
+        **run_options,
+    ) -> EstimatorGradientResult:
+        """Compute the estimator gradients on the given circuits."""
+        jobs, metadata_ = [], []
+        for circuit, observable, parameter_values_, parameters_ in zip(
+            circuits, observables, parameter_values, parameters
+        ):
+            # indices of parameters to be differentiated
+            if parameters_ is None:
+                indices = list(range(circuit.num_parameters))
+            else:
+                indices = [circuit.parameters.data.index(p) for p in parameters_]
+            metadata_.append({"parameters": [circuit.parameters[idx] for idx in indices]})
+
+            offset = np.identity(circuit.num_parameters)[indices, :]
+            plus = parameter_values_ + self._epsilon * offset
+            minus = parameter_values_ - self._epsilon * offset
+            n = 2 * len(indices)
+
+            job = self._estimator.run(
+                [circuit] * n, [observable] * n, plus.tolist() + minus.tolist(), **run_options
+            )
+            jobs.append(job)
+
+        # combine the results
+        if any(job.status() is not JobStatus.DONE for job in jobs):
+            raise QiskitError("The gradient job was not completed successfully. ")
```

Comment:
I think if you call job.result() it waits until a result is available and any failure results in it raising an exception. In Fidelites we ended up wrapping the result() access with a try/catch and re-raising any exception. If here you check status and its still RUNNING on a remote primitive it will fail right since its not done.